### PR TITLE
Added LinkedInput addon.

### PR DESCRIPTION
**Pull request by jimfb**
Added LinkedInput addon.  Goes here until @zpao decides how he wants to package/release it.

### CODE REVIEW

**Comment by jimfb** [comment]
cc @zpao

File: src/addons/link/LinkedInput.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+var React = require('React');
+var LinkedValueUtils = require('LinkedValueUtils');
+
+class LinkedInput extends React.Component {
+  render() {
+    var newprops = Object.assign({}, this.props);
+    newprops.value = LinkedValueUtils.getValue(this.props);
+    newprops.checked = LinkedValueUtils.getChecked(this.props);
+    newprops.valueLink=undefined;
```
Comment (zpao) [inline src/addons/link/LinkedInput.js]: `delete newProps.valueLink` (and yes, `newProps`)

**Comment by jimfb** [comment]
Ping @zpao

**Comment by zpao** [comment]
sure

**Comment by jimfb** [comment]
@zpao said:
 > sure

Are we not using labels anymore?

File: src/addons/link/LinkedInput.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
```
Comment (krzysiek1507) [inline src/addons/link/LinkedInput.js]: 2016 or present?

File: src/addons/link/__tests__/LinkedInput-test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  var React;
+  var ReactDOM;
+
+  beforeEach(function() {
+    LinkedInput = require('LinkedInput');
+    React = require('React');
+    ReactDOM = require('ReactDOM');
+  });
+
+  it('should basically work', function() {
```
Comment (iamdustan) [inline src/addons/link/__tests__/LinkedInput-test.js]: :) this is my kind of test!

File: src/addons/link/__tests__/LinkedInput-test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
```
Comment (krzysiek1507) [inline src/addons/link/__tests__/LinkedInput-test.js]: @jimfb here also `present` ;)

**Comment by zpao** [comment]
> Are we not using labels anymore?
Lazy, stamping quickly. You know what I meant so let's not get too wrapped up in some official process. If the labels actually had any power, I'd use all the time.

But as others mentioned, just fix the `-present` bit.

**Comment by gaearon** [comment]
Has this ever worked? It seems to ignore the change handler on the link.

**Comment by gaearon** [comment]
Fixed this as part of #9385 even though nobody likely ever used this.

**Pull request merged**